### ISSUE TITLE
Failure in running basic_expressions example

### ISSUE THREAD

**Issue by ben-izd** [WolframResearch/wolfram-library-link-rs#44]
[basic_expressions ](https://github.com/WolframResearch/wolfram-library-link-rs/blob/b825fb2b403957c34ae83caa0cf75d1a4f93df61/wolfram-library-link/examples/exprs/basic_expressions.rs) compiles successfully but in Mathematica, as shown in the image, the first arguments get `Echo`ed and then it returns failure. Subsequent calls return the string without any `Echo`ing.

![image](https://user-images.githubusercontent.com/56647066/173227503-155ecf7e-cbe9-4728-9d84-c7bd1c3f9f02.png)

Did I miss something ?

**Comment by ConnorGray** [comment]
The issue here is somewhat subtle, but easy to work around. In your Wolfram code that calls into the library function, you need to include wrapper code that looks like this:

```wolfram
Block[{$Context = "SomeRandomEmptyContext`", $ContextPath = {}},
    yourLibraryFunction[..]
]
```

This forces symbols that are sent across the WSTP link to include their context. Otherwise, symbols in the system context (like [ReturnPacket](https://reference.wolfram.com/language/ref/ReturnPacket)) are sent as the ambiguous symbol name `"ReturnPacket"` instead of the absolute symbol `"System`ReturnPacket"`. If symbols are sent without an explicit context, then the expression parsing code on the Rust side doesn't know how to parse them.

Fixing this issue properly will require a Kernel-level change, which I'm hoping to implement internally in a future WL version. 

For now, I've included helper code in wolfram-library-link to automate away this boilerplate: have a look at the documentation on [creating a loader function for your library](https://docs.rs/wolfram-library-link/latest/wolfram_library_link/fn.exported_library_functions_association.html#creating-a-loader-function).

Using a 'loader function' both eliminates the boilerplate from needing to write your own `LibraryFunctionLoad` calls for every library function, and solves this symbol parsing issue for you.

**Comment by ben-izd** [comment]
Thanks for your rapid response. Your solution work as expected. 
I noticed in the creating loader function page, that there aren't any possible issues when it couldn't determine the library file path (as it did happen to me). It would be more helpful to add a section, even though the name might suggest.

**Comment by ConnorGray** [comment]
Thanks for the suggestion. I will keep this issue open until I've updated the documentation to mention that issue.

To be certain, the error message you got was [this one](https://github.com/WolframResearch/wolfram-library-link-rs/blob/v0.2.5/wolfram-library-link/src/macro_utils.rs#L379-L382), is that correct?

**Comment by ben-izd** [comment]
Yes, that's correct.

**Comment by ConnorGray** [comment]
@ben-izd I've just published v0.2.6 of wolfram-library-link, which includes an additional ["Possible issues" section in the `exported_library_functions_association()` documentation](https://docs.rs/wolfram-library-link/0.2.6/wolfram_library_link/fn.exported_library_functions_association.html#possible-issues) that I hope will clarify the problem you mentioned for others.

Closing this issue now, but please let me know if you have any issues or suggestions for improvements to the new documentation.
